### ISSUE TITLE
deprecate twisted.names.client.ThreadedResolver in favor of twisted.internet.base.ThreadedResolver

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3710]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3710|
|Type|enhancement|
|Created|2009-03-30 16:24:58Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-threadedresolver-3710|

We've basically decided that `twisted.internet.base.ThreadedResolver` is public now, despite the slight unfortunateness that implies.  So we can just point people at that.

Attachments:

* [client.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/595/5952168acbbe69b1f71192a3207de09be589d780/918bab7a34a82ccdc56504d97a38bacd3f9551ca.patch) (807 bytes) - added by cary on 2009-04-02 16:42:29Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3710 3710
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_deprecate_threadedresolver_3710 branches/deprecate-threadedresolver-3710
branch_author__cary__exarkun cary, exarkun
status__closed closed
resolution__fixed fixed
component__names names
keywords__deprecation deprecation
time__1238430298000000 1238430298000000
changetime__1239375431000000 1239375431000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="cary's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|cary removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="cary's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|cary set owner to @dreid|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun|
|-|-|

This should have a test, too.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [26638]) Branching to 'deprecate-threadedresolver-3710'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [26639]) Apply cary's patch to deprecate `ThreadedResolver`

refs [#3710](https://github.com/twisted/twisted/issues/3710)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [26640]) test the deprecation; make it be the right type and point at the right code

refs [#3710](https://github.com/twisted/twisted/issues/3710)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set owner to @exarkun|
|-|-|

Cool, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [26656]) Merge deprecate-threadedresolver-3710

Author: cary, exarkun
Reviewer: therve
Fixes: [#3710](https://github.com/twisted/twisted/issues/3710)

Deprecate `twisted.names.client.ThreadedResolver` in favor of
`twisted.internet.base.ThreadedResolver`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
